components/sections/migrate/synchronize-tab-content.tsx

```diff
@@ -48,11 +53,30 @@ export function SynchronizeTabContent({ onContinue }: SynchronizeTabContentProps
   } = useSynchronization()
 
   const [activeView, setActiveView] = useState<AccountViewType>(AccountViewType.ALL)
+  // Get selection functions from useMigration
+  const { toggleAllAccounts } = useMigration()
 
   const handleMigrate = () => {
     onContinue()
   }
 
+  // Check if all apps are selected
+  const areAllAppsSelected = useMemo(() => {
+    if (appsWithoutErrors.length === 0) return false
+
+    const allAccountsSelected = appsWithoutErrors.every(app => app.accounts?.every(account => account.selected))
+    const allMultisigAccountsSelected = appsWithoutErrors.every(app => app.multisigAccounts?.every(account => account.selected))
+    return allAccountsSelected && allMultisigAccountsSelected
+  }, [appsWithoutErrors])
+
+  // Handle "Select All" checkbox change
+  const handleSelectAllChange = useCallback(
+    (checked: CheckedState) => {
+      toggleAllAccounts(Boolean(checked))
+    },
```

Comment:
CheckedState is a string value, from what I see on other parts of the PR. Are we casting it to boolean on purpose?